src/lib/files.ts

```diff
@@ -116,20 +122,21 @@ export default class FileHandler {
   openFile = async (file: File): Promise<boolean> => {
     this.pages.savePage();
     return this.pages.overwritePages(
-      await JSONReader.read(new AsyncReader(file))
+      await JSONReader.read(AsyncReader.readAsText(file))
     );
   };
 
   private handleImage = async (file: File, cursor): Promise<fabric.Object[]> =>
-    new Promise<fabric.Object[]>((resolve) => {
-      const fileURL = window.URL.createObjectURL(file);
-      fabric.Image.fromURL(fileURL, (obj: fabric.Image) => {
-        resolve(this.pages.canvas.placeObject(obj, cursor));
-      });
-    });
+    new Promise<fabric.Object[]>((resolve) =>
+      AsyncReader.readAsDataURL(file).then((result) =>
+        fabric.Image.fromURL(result.toString(), (obj: fabric.Image) => {
+          resolve(this.pages.canvas.placeObject(obj, cursor));
+        })
+      )
+    );
```

Comment:
Is this preferred?
```ts
private handleImage = async (
  file: File,
  cursor
): Promise<fabric.Object[]> => {
  const elt = document.createElement("img");
  elt.src = (await AsyncReader.readAsDataURL(file)).toString();
  return this.pages.canvas.placeObject(new fabric.Image(elt), cursor);
};
```
I wouldn't think so (why would `fabric.Image.fromURL` exist otherwise, especially since it doesn't do what this does (notice this is synchronous from fabric's POV)) but wanted to see if you knew otherwise